samples/52.teams-messaging-extensions-search-auth-config/src/main/java/com/microsoft/bot/sample/teamssearchauth/TeamsMessagingExtensionsSearchAuthConfigBot.java

```diff
@@ -269,20 +269,23 @@ public TeamsMessagingExtensionsSearchAuthConfigBot(
     ) {
         if (action.getCommandId().toUpperCase().equals("SIGNOUTCOMMAND")) {
             UserTokenProvider tokenProvider = (UserTokenProvider) turnContext.getAdapter();
+
+            TaskModuleTaskInfo taskInfo = new TaskModuleTaskInfo();
+            taskInfo.setCard(createAdaptiveCardAttachment());
+            taskInfo.setHeight(200);
+            taskInfo.setWidth(400);
+            taskInfo.setTitle("Adaptive Card: Inputs");
+
+            TaskModuleContinueResponse continueResponse = new TaskModuleContinueResponse();
+            continueResponse.setValue(taskInfo);
+
+            MessagingExtensionActionResponse actionResponse = new MessagingExtensionActionResponse();
+            actionResponse.setTask(continueResponse);
             return tokenProvider.signOutUser(
                 turnContext,
                 connectionName,
                 turnContext.getActivity().getFrom().getId()
-            ).thenApply(response -> new MessagingExtensionActionResponse() {{
-                setTask(new TaskModuleContinueResponse() {{
-                    setValue(new TaskModuleTaskInfo() {{
-                        setCard(createAdaptiveCardAttachment());
-                        setHeight(200);
-                        setWidth(400);
-                        setTitle("Adaptive Card: Inputs");
-                    }});
-                }});
-            }});
+            ).thenApply(response -> actionResponse);
```

Comment:
Same as before, this should be inside the `thenApply`

Reply:
I'm not sure with this one, I think that the creation of the object shouldn't be inside the return statement with this approach

Reply:
Before DBI -> SI, the DBI was created inside the `thenApply`.

After the change, we are doing this outside, so this is a disparity